### ISSUE TITLE
Not stable behaviour of "stdout" + "sprintf"

### ISSUE THREAD

**Issue by MikhailLipanin** [objectionary/eo#1323]
I have a simple program:
```
[] > fff
  QQ.io.stdout > @
    QQ.txt.sprintf
      "out is %d\n"
      123
```
then I run:
```
eoc clean
eoc dataize fff
```
and I get the following error:
```
EOorg.EOeolang.EOstringν12="Error at "EOorg.EOeolang.EOio.EOstdout#text" attribute; caused by Can't #copy() package object 'org.eolang.txt.sprintf'"SF
```
BUT, if I run `eoc clean` again, and adding aliases to my code, so the code will looks like:
```
+alias org.eolang.txt.sprintf
+alias org.eolang.io.stdout

[] > fff
  stdout > @
    sprintf
      "out is %d\n"
      123
```
and then `eoc dataize fff`  -> All is working OK.
@yegor256 , I think there is something wrong with uploading objects

**Comment by mximp** [comment]
@yegor256 so here we have fundamental issue: there is no way to distinguish between package (or method) and object with current structure of parsing and building. It's incorporated in parser and runtime.
The limitation (or rule) may sound like this: **_we cannot reference fully qualified names of objects within EO sources unless there is `+rt` meta containing those objects_**

**Comment by yegor256** [comment]
@mximp let's create a new XSL, which will go through all code (before `QQ.io.stdout` gets expanded to four `<o/>` elements (`Q .org .io .stdout`) and add necessary `+alias` items to the `<head>` of XMIR. WDYT?

**Comment by mximp** [comment]
@yegor256 I guess in this case we will get exception about unused aliases. As I understand if we supply alias for an object it should be mentioned just by name within the code (not by fully qualified name). So if XSL add alias it should also leave only `stdout` part in the object itself.

**Comment by yegor256** [comment]
@mximp yes, good idea. We just strip `QQ.txt.sprintf` to `x24242` and add `+alias x24242 org.eolang.txt.sprintf`. I'm saying `x24242` to avoid collisions between aliases. Maybe there is already `sprintf`.

**Comment by mximp** [comment]
@yegor256 There is a pitfall in our plan )).. There is no way we can distinguish an object reference from its attribute.
Suppose we have a reference in the source: `org.eolang.txt.super-sprintf.attr`. We cannot automatically identify what we should alias here: is `attr` an object? or `super-sprintf`?. At the moment it is solved at runtime walking though all parts  starting from `org`, `eolang`, etc. For each object the Runtime will try to find Java class and if it's found corresponding object is returned otherwise it considers the part is a package and continue traversal.  
At XMIR level we don't have such opportunity.   
Can we just mandate to use either `+alias` or `+rt` metas? Maybe introduce parsing waring if fully qualified names are used and there is no  `+rt` meta?

**Comment by yegor256** [comment]
@mximp how about this:

```
QQ.io.stdout
  "Hello"
```

we turn into:

```
+alias org
+alias org.eolang
+alias org.eolang.io
+alias org.eolang.io.stdout

QQ.io.stdout
  "Hello"
```

**Comment by mximp** [comment]
@yegor256 All 4 aliases will result in `unused alias` error, since we don't use any of them. They will be expanded into `org org`, `eolang org.eolang`, `io org.eolang.io`, `stdout org.eolang.io.stdout` which is never used in the code.

**Comment by yegor256** [comment]
@mximp why can't do this:

```
QQ.io.stdout.xyz
  "Hello"
```

to 

```
+alias xyz org.eolang.io.stdout.xyz

xyz
  "Hello"
```

**Comment by yegor256** [comment]
@mximp how about this. We turn this code:

```
QQ.io.stdout.xyz
  "Hello"
```

into this:

```
+foreign org
+foreign org.eolang
+foreign org.eolang.stdout
+foreign org.eolang.stdout
+foreign org.eolang.stdout.xyz

org.eolang.io.stdout.xyz
  "Hello"
```

This `+foreign x` meta will tell `ProbeMojo` (we will create it): "please, try to find _x_ in Objectionary, if you fail, just ignore it. If you succeed, put _x_ into `foreign.csv` just like `DiscoverMojo` does."

WDYT?

**Comment by mximp** [comment]
@yegor256 that might work. Let's stick with this.

**Comment by 0pdd** [comment]
@MikhailLipanin the puzzle [#1395](https://github.com/objectionary/eo/issues/1395) is still not solved.

**Comment by 0pdd** [comment]
@MikhailLipanin 3 puzzles [#1677](https://github.com/objectionary/eo/issues/1677), [#1678](https://github.com/objectionary/eo/issues/1678), [#1679](https://github.com/objectionary/eo/issues/1679) are still not solved; solved: [#1395](https://github.com/objectionary/eo/issues/1395).

**Comment by 0pdd** [comment]
@MikhailLipanin 2 puzzles [#1677](https://github.com/objectionary/eo/issues/1677), [#1678](https://github.com/objectionary/eo/issues/1678) are still not solved; solved: [#1395](https://github.com/objectionary/eo/issues/1395), [#1679](https://github.com/objectionary/eo/issues/1679).

**Comment by 0pdd** [comment]
@MikhailLipanin the puzzle [#1700](https://github.com/objectionary/eo/issues/1700) is still not solved; solved: [#1395](https://github.com/objectionary/eo/issues/1395), [#1677](https://github.com/objectionary/eo/issues/1677), [#1678](https://github.com/objectionary/eo/issues/1678), [#1679](https://github.com/objectionary/eo/issues/1679).

**Comment by 0pdd** [comment]
@MikhailLipanin all 5 puzzles are solved here: [#1395](https://github.com/objectionary/eo/issues/1395), [#1677](https://github.com/objectionary/eo/issues/1677), [#1678](https://github.com/objectionary/eo/issues/1678), [#1679](https://github.com/objectionary/eo/issues/1679), [#1700](https://github.com/objectionary/eo/issues/1700).

**Comment by MikhailLipanin** [comment]
this issue can be closed, since the problem is solved
